src/agent/onefuzz/src/syncdir.rs

```diff
@@ -26,13 +26,17 @@ const DEFAULT_CONTINUOUS_SYNC_DELAY_SECONDS: u64 = 60;
 #[derive(Debug, Deserialize, Clone, PartialEq)]
 pub struct SyncedDir {
     pub path: PathBuf,
-    pub url: BlobContainerUrl,
+    pub url: Option<BlobContainerUrl>,
 }
 
 impl SyncedDir {
     pub async fn sync(&self, operation: SyncOperation, delete_dst: bool) -> Result<()> {
+        if self.url.is_none() {
+            bail!("only able to sync with remote URLs");
```

Comment:
should it not be a no op instead of failing. 
This could allows the code to be agnostic of the type storage used here